scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -823,8 +821,16 @@ class FormatOps(val tree: Tree, val initStyle: ScalafmtConfig) {
       if (isBracket) close // If we can fit the type params, make it so
       else lastParen // If we can fit all in one block, make it so
 
+    val arity = valueParamsOwner match {
+      case d: Decl.Def => d.paramss.foldLeft(0)(_ + _.size)
```

Comment:
Missing `Defn.Def`, `Defn.Macro`, `Ctor.Secondary` and `Init`

Scalameta trees are quite precise, which is sometimes annoying but other times a blessing

Reply:
Also, should this rule affect call-site? In that case you need to extract from `Term.Apply`.

Reply:
No IMO it shouldn't affect the call-site

Reply:
I can't find `Init` anywhere? :)